testing/internal/e2e/tests/base/scope_permissions_test.go

```diff
@@ -0,0 +1,165 @@
+	"github.com/hashicorp/boundary/api/scopes"
+	"github.com/hashicorp/boundary/testing/internal/e2e"
+	"github.com/hashicorp/boundary/testing/internal/e2e/boundary"
+	"github.com/stretchr/testify/require"
+)
+
+// TestScopePermissions tests an issue with listing scopes
+// When a user is only given permission to only list single project in a scope,
+// the user should only see that project and no error should be displayed
+func TestScopePermissions(t *testing.T) {
+	e2e.MaybeSkipTest(t)
+	bc, err := boundary.LoadConfig()
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	boundary.AuthenticateAdminCli(t, ctx)
+
+	// Create Org and 2 Projects
+	orgId, err := boundary.CreateOrgCli(t, ctx)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		ctx := context.Background()
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary", e2e.WithArgs("scopes", "delete", "-id", orgId))
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+	projectId, err := boundary.CreateProjectCli(t, ctx, orgId)
+	require.NoError(t, err)
+	_, err = boundary.CreateProjectCli(t, ctx, orgId)
+	require.NoError(t, err)
+
+	// Create new account
+	const testAccountName = "test-account"
+	accountId, acctPassword, err := boundary.CreateAccountCli(t, ctx, bc.AuthMethodId, testAccountName)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary",
+			e2e.WithArgs("accounts", "delete", "-id", accountId),
+		)
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+
+	// Create new user
+	userId, err := boundary.CreateUserCli(t, ctx, "global")
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		boundary.AuthenticateAdminCli(t, ctx)
+		output := e2e.RunCommand(ctx, "boundary",
+			e2e.WithArgs("users", "delete", "-id", userId),
+		)
+		require.NoError(t, output.Err, string(output.Stderr))
+	})
+	err = boundary.SetAccountToUserCli(t, ctx, userId, accountId)
+	require.NoError(t, err)
+
+	// Find Ids of the roles to be modified
+	output := e2e.RunCommand(ctx, "boundary", e2e.WithArgs("roles", "list", "-format", "json"))
+	var rolesListResult roles.RoleListResult
+	err = json.Unmarshal(output.Stdout, &rolesListResult)
+	require.NoError(t, err)
+	var loginGrantsRoleId string
+	var authenticatedUserGrantsRoleId string
+	var loginAndDefaultGrantsRoleId string
+	for _, item := range rolesListResult.Items {
+		if item.Name == "Login Grants" {
+			loginGrantsRoleId = item.Id
+		}
+		if item.Name == "Authenticated User Grants" {
+			authenticatedUserGrantsRoleId = item.Id
+		}
+	}
+	output = e2e.RunCommand(ctx, "boundary", e2e.WithArgs("roles", "list", "-format", "json", "-scope-id", orgId))
+	err = json.Unmarshal(output.Stdout, &rolesListResult)
+	require.NoError(t, err)
+	for _, item := range rolesListResult.Items {
+		if item.Name == "Login and Default Grants" {
+			loginAndDefaultGrantsRoleId = item.Id
+		}
+	}
+
+	// Modify login and default grants role
```

Comment:
Maybe we should add context here as to what we're doing with these modifications.
- In the org, prevent the user from being able to list all scopes